### PR TITLE
Stable-order pagination for open GitHub pull request listing

### PR DESCRIPTION
**Pull request by krlvi**
Fixes [GB-1884](https://linear.app/gitbutler/issue/GB-1884/list-reviews-fails-on-active-repos-with-100-prs-open-pull-request).

`list_reviews` failed on active repos with 100+ open PRs: the listing was fetched under `sort=updated`, which reshuffles on any PR activity, and the drift-verification loop required two identical full snapshots before returning — active repos could never settle, so the refresh failed outright with "Open pull request listing changed while paginating".

Fetch sorted by creation time ascending instead, which makes offset pagination stable: updates never move a PR and new PRs only append. A single scan then suffices, so the drift-verification loop is deleted and API traffic for large repos is halved. A close or reopen racing the scan can still shift a page boundary — duplicates keep their freshest copy, and an omitted PR returns on the next refresh.

The returned listing stays most-recently-updated first: desktop, lite, and TUI consumers pick one review per source branch positionally and expect the freshest PR to win, so the fetch order stays an internal detail.

Also fixes a latent mock-server race in the pagination tests: accepted streams inherited non-blocking from the listener and panicked on WouldBlock reads.

### CODE REVIEW

File: crates/but-github/src/client.rs [hunk 1]
```diff
@@ -247,43 +249,35 @@ impl GitHubClient {
...
-            }
-        }
+        let pulls = self
+            .get_all_pages::<GitHubPullRequest>(&url, MAX_PAGES + 1)
+            .await?;
+        // Later duplicates overwrite earlier ones, keeping the freshest copy.
+        let pulls: std::collections::BTreeMap<_, _> =
+            pulls.into_iter().map(|pull| (pull.number, pull)).collect();
+        anyhow::ensure!(
+            pulls.len() <= MAX_PAGES * 100,
```
[the remark this answers is not in the document]
Comment (krlvi) [reply crates/but-github/src/client.rs]: Reworked in d8cff220c6e: the overflow check now runs on the raw pre-dedupe count, testing whether all 101 fetched pages were full — the precise signal that the listing may extend past the bound. Note the original scenario as stated (short page 101 with a boundary duplicate) is a repo with exactly 10,000 open PRs, where the deduplicated listing is complete rather than truncated — the new check accepts that case while still failing when 101 full pages leave real overflow possible.

File: crates/but-github/src/client.rs [hunk 2]
```diff
@@ -247,43 +249,35 @@ impl GitHubClient {
...
+            "Open pull request listing exceeded {MAX_PAGES} pages"
+        );
```
[the remark this answers is not in the document]
Comment (krlvi) [reply crates/but-github/src/client.rs]: Fixed in d8cff220c6e: the check now runs on the raw fetched count before deduplication, firing when all 101 pages were full. A short page 101 completes the listing (including the duplicate-at-the-boundary case, where the deduplicated result is exact), so the bound can no longer be defeated by duplicates nor false-fail a complete listing; tests updated to pin both sides.

File: crates/but-github/src/client.rs [hunk 3]
```diff
@@ -247,43 +249,35 @@ impl GitHubClient {
         Ok(response)
     }
 
-    /// Fetch the list of the open PRs on a repo.
+    /// Fetch the list of the open PRs on a repo, most recently updated first.
+    ///
+    /// Fetched sorted by creation time ascending so offset pagination stays
```
[the remark this answers is not in the document]
Comment (krlvi) [reply crates/but-github/src/client.rs]: Fixed in d8cff220c6e — reworded to "Pages are fetched in ascending creation-time order …".
